packages/connect-form/src/components/advanced-options-tabs/authentication-tab/authentication-default.tsx

```diff
@@ -1,9 +1,181 @@
-import React from 'react';
+import React, { useCallback } from 'react';
+import {
+  Icon,
+  IconButton,
+  Label,
+  RadioBox,
+  RadioBoxGroup,
+  TextInput,
+  css,
+  spacing,
+} from '@mongodb-js/compass-components';
+import ConnectionStringUrl from 'mongodb-connection-string-url';
+import { AuthMechanism } from 'mongodb';
+
+import { UpdateConnectionFormField } from '../../../hooks/use-connect-form';
+import FormFieldContainer from '../../form-field-container';
+import { ConnectionFormError } from '../../../utils/validation';
+
+const authSourceLabelStyles = css({
+  padding: 0,
+  margin: 0,
+  flexGrow: 1,
+});
+
+const infoButtonStyles = css({
+  verticalAlign: 'middle',
+  marginTop: -spacing[1],
+});
+
+const defaultAuthMechanismOptions: {
+  title: string;
+  value: AuthMechanism;
+}[] = [
+  {
+    title: 'Default',
+    value: AuthMechanism.MONGODB_DEFAULT,
+  },
+  {
+    title: 'SCRAM-SHA-1',
+    value: AuthMechanism.MONGODB_SCRAM_SHA1,
+  },
+  {
+    title: 'SCRAM-SHA-256',
+    value: AuthMechanism.MONGODB_SCRAM_SHA256,
+  },
+];
+
+function AuthenticationDefault({
+  errors,
+  connectionStringUrl,
+  updateConnectionFormField,
+}: {
+  connectionStringUrl: ConnectionStringUrl;
+  errors: ConnectionFormError[];
+  updateConnectionFormField: UpdateConnectionFormField;
+}): React.ReactElement {
+  const password = connectionStringUrl.password;
+  const username = connectionStringUrl.username;
+
+  const selectedAuthMechanism =
+    connectionStringUrl.searchParams.get('authMechanism') ?? '';
+  const selectedAuthTab =
+    defaultAuthMechanismOptions.find(
+      ({ value }) => value === selectedAuthMechanism
+    ) ?? defaultAuthMechanismOptions[0];
+
+  const onAuthMechanismSelected = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      event.preventDefault();
+      updateConnectionFormField({
+        type: 'update-search-param',
```

Comment:
Currently this action also sets empty values, ie. it would set `&authMechanism=&` for a null value, could we also add `updatedSearchParams.delete(action.currentKey);` in case the value is empty here https://github.com/mongodb-js/compass/blob/main/packages/connect-form/src/hooks/use-connect-form.ts#L386-L393?

Reply:
I see how that helper could be useful here, but I think we want to allow empty values to be set since we use this in the advanced tab to help populate the advanced options. In that case setting something to empty is useful as part of the process of entering those options:
<img width="716" alt="Screen Shot 2022-01-24 at 1 43 41 PM" src="https://user-images.githubusercontent.com/1791149/150844594-db516380-babb-486f-9706-a1a5d22de4d2.png">

Reply:
I'm not sure that is so useful though, there is no search param that makes sense as empty, and I don't know why someone would prefer to populate the value in the connection string rather than in the form after they selected the key there. I believe is a good feature if we produce a cleaner connection string without random empty search params. 

Do you have an alternative to at least delete `authMechanism` and the other params if they are empty?

Reply:
On this page the `authMechanism` will always be set since it's a radio with 3 options and unsetting isn't one.
Yup we can use the `delete-search-param` action.
I see what you mean. I'm cool with either in that case. Not sure this is the pr for that change.